storage/redis/peer_store.go

```diff
@@ -267,28 +239,74 @@ type peerStore struct {
 	wg     sync.WaitGroup
 }
 
-func (ps *peerStore) groups() []string {
-	return []string{bittorrent.IPv4.String(), bittorrent.IPv6.String()}
-}
+var addressFamilies = [2]string{"4", "6"}
 
-func (ps *peerStore) leecherInfohashKey(af, ih string) string {
-	return af + "_L_" + ih
-}
+// Key that stores the peers for an InfoHash.
+//
+// "4S0102030405060708090a0b0c0d0e0f1011121314" => IPv4 Seeder with Peer ID 0102030405060708090a0b0c0d0e0f1011121314
+// "6L0204090405060708060a0b0c0d0e0f1011121328" => IPv6 Seeder with Peer ID 0204090405060708060a0b0c0d0e0f1011121328
+func swarmKey(infoHash bittorrent.InfoHash, seed bool, ip netaddr.IP) string {
+	var b strings.Builder
+	switch {
+	case ip.Is4(), ip.Is4in6():
+		b.Grow(1 + 1 + 4) // "4" + "S"/"L" + IPv4
```

Comment:
Missing 20 bytes for the actual infohash 🙃

Reply:
nice catch

I also noticed that in PutLeecher it does not check if the infohash was new. It's apparently missing a 

```go
	if reply[1] == 1 { // The Infohash Key (and thus the infohash) was new.
		_, err = conn.Do("INCR", infohashCountKey(ip))
		if err != nil {
			return err
		}
	}
```

Is that on purpose? Should I add this?

Reply:
I believe that's on purpose, because for redis we only count infohashes with at least one seeder. I don't remember why I did it like that, but I assume there was some reason for it...maybe it would have been difficult to correctly determine whether a seeder-only or leecher-only swarm already exists or something, not sure.